### PR TITLE
Ensure passthrough clusters can be created

### PR DESCRIPTION
**Pull request by freddygv**
Previously their creation was gated on the existence of the global mesh
config. The associated listener filter chain was being created
correctly.

### CODE REVIEW

File: agent/xds/clusters_test.go [hunk 1]
```diff
@@ -640,6 +640,14 @@ func TestClustersFromSnapshot(t *testing.T) {
 		{
 			name:   "transparent-proxy",
 			create: proxycfg.TestConfigSnapshot,
+			setup: func(snap *proxycfg.ConfigSnapshot) {
+				snap.Proxy.Mode = structs.ProxyModeTransparent
+				snap.ConnectProxy.MeshConfigSet = true
```
Comment (freddygv) [inline agent/xds/clusters_test.go:645]: The cluster should be created even if the config is nil.

**Review by rboyer — APPROVED** [review]
LGTM

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/375709.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 353280660fb49aa195244329ac11e0aa4b4eaf25 onto `release/1.10.x` succeeded!

**Pull request merged**